lib/PreferenceList.cc

```diff
@@ -47,6 +48,40 @@ PrefListElement PreferenceList::at(SizeType index) const {
     return pref_list_.at(index);
 }
 
+void PreferenceList::set_ties(RankType rank, PreferenceList p){
+    SizeType size = p.size();
+    cur_rank_++;
+    is_tied_[rank] = true;
+    for(SizeType i=0; i<size; i++){
```

Comment:
for (auto i = 0; i < size; ++i) {
Appropriate spacing in for loops, and use of auto so compiler can auto deduce type of i.